cpu/native/syscalls.c

```diff
@@ -134,6 +134,10 @@ void _native_syscall_leave(void)
     }
 }
 
+/* make use of TLSF if it is included, except when building with valgrind
+ * support, where one probably wants to make use of valgrind's memory leak
+ * detection abilities*/
+#if !(defined MODULE_TLSF) || (defined(HAVE_VALGRIND_H))
```

Comment:
To me it looks like `free` needs exactly the same guard (and all other related functions as well: realloc, calloc, memallign). Otherwise TLSF pointer will be freed by glib, which should fail violently.